lifelines/utils/sklearnAdapter.py

```diff
@@ -0,0 +1,84 @@
+import inspect
+import pandas
+from sklearn.base import BaseEstimator, RegressorMixin, MetaEstimatorMixin, MultiOutputMixin
```

Comment:
scikit learn isn't a dependency for the project, and I don't want to natively make it one. Can we do a try-catch import here, with an error message saying to install scikit-learn?

Reply:
This module is not imported automatically when importing `lifelines.utils` or `from lifelines.utils import *` because it is not in `__all__`. And it is useful only for the ones already using `sklearn`. So if a person imports this module he is expected to have `sklearn`. Otherwise he just doesn't import it and `sklearn` is not needed;